src/metric/definition.go

```diff
@@ -727,9 +727,8 @@ var KSMSpecs = definition.SpecGroups{
 		},
 	},
 	"hpa": {
-		IDGenerator:     prometheus.FromLabelValueEntityIDGenerator("kube_hpa_labels", "hpa"),
-		TypeGenerator:   prometheus.FromLabelValueEntityTypeGenerator("kube_hpa_labels"),
-		NamespaceGetter: prometheus.FromLabelGetNamespace,
+		IDGenerator:   prometheus.FromLabelValueEntityIDGenerator("kube_hpa_labels", "hpa"),
+		TypeGenerator: prometheus.FromLabelValueEntityTypeGenerator("kube_hpa_labels"),
```

Comment:
horizontalpodautoscalers          hpa          autoscaling/v1                         true         HorizontalPodAutoscaler

Reply:
I've added the filtering because HPA are indeed namespaced as Paolo pointed out from the kubectl api-resources call.